src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/ConnectionSetupDistributedTracing.cs

```diff
@@ -25,11 +25,12 @@ internal static class ConnectionSetupDistributedTracing
 
             if (activity is not null)
             {
-                activity.DisplayName = $"HTTP connection_setup {authority.HostValue}:{authority.Port}";
+                Debug.Assert(serverAddress is not null, "serverAddress should not be null when System.Net.Http.EnableActivityPropagation is true.");
```

Comment:
In fact, we should be outside of the call and `serverAddress` should be non-nullable, see #114326. If we decide to fix this we can do it right in the PR or in a follow-up.